server/routes/api/team.ts

```diff
@@ -60,4 +62,65 @@ router.post("team.update", auth(), async (ctx) => {
   };
 });
 
+router.post("team.create", auth(), async (ctx) => {
+  const { user } = ctx.state;
+  const { name } = ctx.body;
+  assertLength(name, 2, "Name must be 2 or more characters");
+
+  const existingTeam = await Team.scope("withAuthenticationProviders").findByPk(
+    user.teamId
+  );
+
+  authorize(user, "createTeam", existingTeam);
+
+  const authenticationProviders = existingTeam?.authenticationProviders.map(
+    (provider) => {
+      return {
+        name: provider.name,
+        providerId: provider.providerId,
+      };
+    }
+  );
+
+  invariant(
+    authenticationProviders?.length,
+    "authentication provideers must exist"
```

Comment:
```suggestion
    "authentication providers must exist"
```

Reply:
Interesting though though, do they _need_ to in the long run? Wouldn't that just mean a team with magic link only signin?

Reply:
yes that's true; for the time being I am treating users without auth provider as "guests"